### ISSUE TITLE
Hg38 VCF file error: Error in UseMethod("select") :    no applicable method for 'select' applied to an object of class "NULL"

### ISSUE THREAD

**Issue by nerea-bilbao** [honzee/RNAseqCNV#19]
Hi everyone,

I am trying to execute a leukemia sample aligned with STAR with hg38, featurecounts (hg38.gtf) and GATK Haplotype caller inputs. 

Count header:
ENSG00000160072	1222
ENSG00000279928	87
ENSG00000228037	37

VCF header:
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	851
chr1	10560	.	C	G	37.32	.	AC=2;AF=1.00;AN=2;DP=2;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;QD=18.66;SOR=0.693	GT:AD:DP:GQ:PL	1/1:0,2:2:6:49,6,0
chr1	14464	.	A	T	152.14	.	AC=2;AF=1.00;AN=2;DP=4;ExcessHet=0.0000;FS=0.000;MLEAC=2;MLEAF=1.00;MQ=60.00;QD=25.36;SOR=3.258	GT:AD:DP:GQ:PL	1/1:0,4:4:12:166,12,0

The command:
RNAseqCNV_wrapper(config = "C:/Users/neret/Documents/R/win-library/4.1/RNAseqCNV/extdata/config_mock_2.txt", metadata = "C:/Users/neret/Documents/R/win-library/4.1/RNAseqCNV/extdata/metadata_mock_851.csv", snv_format = "vcf", genome_version = "hg38")

And I obtain this answer which I read in the #7 Issue referring but, the programe has been updated for hg38, so I do not understand why is happening this. I guess it is something related to the VCF file:

Error in UseMethod("select") : 
  no applicable method for 'select' applied to an object of class "NULL"

Thanks in advance,

**Comment by honzee** [comment]
Hi,

I am glad you are trying out our package!

Unfortunately, the error message is very unspecific, so there are many possibilities which might be causing this. If there is any other information regarding the error, please send it as well!
RNAseqCNV_wrapper generates messages, which show which part of the analysis is running. This message will also help us locate the issue.

Is this happening for all of your files, or only for certain ones?

The two most probable causes are:
1. Incompatible input - the format of the count file and the vcf file look good at first glance. This does not necessarily mean there is no other issue with them. The metadata and config files might also be incompatible. The fastest way to troubleshoot would be to send me all the input files, and I could analyze what is happening in detail, but I understand, that this might not be possible due to data security.
2. Versioning problem - although I think this is less likely due to the type of your error, it might be the case, that due to newer versions of some packages, RNAseqCNV does not run as expected. You could try running your analysis in a docker container if you want to test this.

In summary, please send me all the information regarding the error, so we can be more specific in troubleshooting.

Best,
Jan

**Comment by nerea-bilbao** [comment]
Dear Honzee,

Thank you very much for your dedication! 

By this "RNAseqCNV_wrapper generates messages, which show which part of the analysis is running. This message will also help us locate the issue" you mean this message?

[1] "Analysis initiated"
[1] "Normalization for sample: 851 completed"
[1] "Preparing file with snv information for: 851"
Reading in vcf file..
Extracting depth..
Extracting reference allele and alternative allele depths..
Needed information from vcf extracted
Finished reading vcf
Error in UseMethod("select") : 
  no applicable method for 'select' applied to an object of class "NULL"

I would be glad to send you the inputs to your email for the troubleshooting.

Thanks a lot for you time and help,

Best regards,

Nere,

**Comment by honzee** [comment]
I see, thank you for the information. If you are ok with this, you can send me the files to: jan.barinka@charite.de

**Comment by honzee** [comment]
Hi Nere,

I have looked into your files and there was an issue with the name of the sample. RNAseqCNV did not assume, that the file name could be purely numerical (like 851), which subsequently caused an error when creating a name for an element in a list. I corrected this bug and pushed the changes to the repository. You can either download the new version or not use numerical sample names.

Thank you for letting me know about this and sorry for the trouble.

Best,
Jan

**Comment by nerea-bilbao** [comment]
Dear Honzee,

I am very glad to have received your email and what's more, checking that
my samples run you program and that I was able to confirm hyperdiploidy.

Thank you for your help and dedication,

Great program,

Best regards,

Nere,

**Comment by honzee** [comment]
That makes me quite happy, thank you. 

I will be closing the issue now.
